src/tribol/physics/Mortar.cpp

```diff
@@ -631,6 +641,565 @@ void ComputeSingleMortarJacobian( SurfaceContactElem& elem )
   return;
 }
 
+#ifdef TRIBOL_USE_ENZYME
+
+//------------------------------------------------------------------------------
+int ApplyNormalEnzyme( CouplingScheme* cs )
+{
+  auto planes_view = cs->get3DContactPlanes().view();
+  auto& lm_opts = cs->getEnforcementOptions().lm_implicit_options;
+  bool compute_jacobian = false;
+  if ( lm_opts.eval_mode == ImplicitEvalMode::MORTAR_RESIDUAL_JACOBIAN ||
+       lm_opts.eval_mode == ImplicitEvalMode::MORTAR_JACOBIAN ) {
+    if ( lm_opts.sparse_mode == SparseMode::MFEM_ELEMENT_DENSE ) {
+      cs->getMethodData()->reserveBlockJ(
+          { BlockSpace::NONMORTAR, BlockSpace::MORTAR, BlockSpace::LAGRANGE_MULTIPLIER }, planes_view.size() );
+      cs->createNormalJacobian();
+      cs->getdnMethodData()->reserveBlockJ(
+          { BlockSpace::NONMORTAR, BlockSpace::MORTAR, BlockSpace::LAGRANGE_MULTIPLIER }, planes_view.size() );
+      compute_jacobian = true;
+    } else {
+      SLIC_WARNING( "Unsupported Jacobian storage method." );
```

Comment:
Is this slic warning an issue with an enzyme callable function?

Reply:
No, I only focused on getting this working with the MFEM interface for now, since that's what Serac uses.